dvc/repo/ls.py

```diff
@@ -66,7 +69,15 @@ def onerror(exc):
 
     ret = {}
     for info in infos:
-        metadata = fs.metadata(info)
+        try:
+            metadata = fs.metadata(info)
+        except FileNotFoundError:
+            if dvc_only:
+                raise
```

Comment:
With the current approach (creating empty Metadata) I think it makes sense to remove this check.
it shouldn't matter if we use `dvc_only` or not. In my initial version lack of this was a bug, silently skipping the error, but now it seems ok to remove it. Users should not be punished for a specific check. If there is a broken symlink, other commands will make sure of erroring out

Reply:
Also, in the case of `dvc_only` files, we should never get here: repo_fs will be able to get the metadata even for the removed file